src/network/core/address.h

```diff
@@ -75,16 +73,13 @@ class NetworkAddress {
 	 * @param port the port
 	 * @param family the address family
 	 */
-	NetworkAddress(const char *hostname = "", uint16 port = 0, int family = AF_UNSPEC) :
+	NetworkAddress(std::string_view hostname = "", uint16 port = 0, int family = AF_UNSPEC) :
 		address_length(0),
 		resolved(false)
 	{
-		/* Also handle IPv6 bracket enclosed hostnames */
-		if (StrEmpty(hostname)) hostname = "";
-		if (*hostname == '[') hostname++;
-		strecpy(this->hostname, StrEmpty(hostname) ? "" : hostname, lastof(this->hostname));
-		char *tmp = strrchr(this->hostname, ']');
-		if (tmp != nullptr) *tmp = '\0';
+		if (!hostname.empty() && hostname.front() == '[') hostname.remove_prefix(1);
+		if (!hostname.empty() && hostname.back() == ']') hostname.remove_suffix(1);
```

Comment:
this always bothered me, and has nothing to do with your PR, so feel free to ignore it:

shouldn't it check if there is a `[` and `]` around the text and remove both? Instead of first one then the other? Feels so super weird ..